src/Provider/AbstractProvider.php

```diff
@@ -323,17 +321,12 @@ protected function fetchProviderData($url)
         try {
             $client = $this->getHttpClient();
 
-            $options = [];
+            $httpResponse = $client->get($url, $this->headers ?: []);
 
-            if ($this->headers) {
-                $options['headers'] = $this->headers;
-            }
-
-            $request = $client->get($url, $options);
-            $response = $request->getBody();
-        } catch (BadResponseException $e) {
+            $response = (string) $httpResponse->getBody();
+        } catch (HttpAdapterException $e) {
             // @codeCoverageIgnoreStart
-            $raw_response = explode("\n", $e->getResponse());
+            $raw_response = explode("\n", (string) $e->getResponse()->getBody());
```

Comment:
I believe this has changed in recent 0.x code. I need to merge changes from master into the 1.0 branch, and then we'll see if there are any conflicts.